openhtf/__init__.py

```diff
@@ -394,6 +394,33 @@ def with_args(self, **kwargs):
     new_info.measurements = [m.with_args(**kwargs) for m in self.measurements]
     return new_info
 
+  def with_plugs(self, **subplugs):
+    """Substitute plugs for placeholders for this phase."""
+    # Make a copy so we can have multiple of the same phase with different plugs
+    # in the same test.
+    new_info = mutablerecords.CopyRecord(self)
+    
+    for substitute_name, substitute_class in subplugs.iteritems():
+      # self.plugs contains PhasePlug so enumerate and unpack
+      for i, original_plug in enumerate(self.plugs):
```

Comment:
i'd be okay with that solution too... the point wasn't the performance hit, it was the structure of doing two nested loops to perform what is essentially a set intersection.  any restructure that removes the need for the second loop is fine :)